mlir/include/Quantum/IR/QuantumInterfaces.td

```diff
@@ -71,6 +71,18 @@ def QuantumGate : OpInterface<"QuantumGate"> {
             "Return all results which are considered output qubit values.",
             "mlir::ValueRange", "getQubitResults"
         >,
+        InterfaceMethod<
+            "Return all operands which are considered controlling input qubit values.",
+            "mlir::ValueRange", "getCtrlQubitOperands"
+        >,
+        InterfaceMethod<
+            "Return all operands which are considered controlling input boolean values.",
+            "mlir::ValueRange", "getCtrlValueOperands"
+        >,
+        InterfaceMethod<
+            "Return all operands which are considered controlling output qubit values.",
+            "mlir::ValueRange", "getCtrlQubitResults"
+        >,
```

Comment:
The first thing I would suggest is to respect the original definition of this interface, so `getQubitOperands` should really return _all_ qubits acted on by the device, and same for the results. This simplifies our transformation code in many cases where we don't care about the specifics (this wouldn't need a change here but where the interface methods are implemented).

The second suggestion I have, is that if we want a specific way of interacting with controllable operations, is to create another interface for it, similar to how we have a `ParametrizedGate` and `DifferentiableGate` interface. I'm not 100% convinced we need this, since it sort of depends on whether you consider control wires and values to be a fundamental property of gates, or just a certain subset of gates. I think it might be neater to split them into two interfaces (`ControlledGate` can inherit from `QuantumGate`), but I'll leave this one up to you.

Reply:
> Respect the original definition of this interface, so getQubitOperands

Done

> to create another interface for it, similar to how we have a ParametrizedGate and DifferentiableGate interface

My thoughts here is that we don't actually need a controlled interface more that the interface for operation-arithmetics. In some sense, I think that all linear gates should be controlled so it is fine to just update the QuantumGate interface.

Reply:
Sounds good, thanks :)